### ISSUE TITLE
Dockerization for Ease of Use

### ISSUE THREAD

**Issue by umer2001** [hariprasanths/synchly#24]
**Objective**
This issue aims to Dockerize the Synchly tool to eliminate the need for users to install system-level dependencies such as pg_dump, mongo_dump, and others. By containerizing the tool, users only need to run the container, mount a volume, create configuration files, and seamlessly utilize the tool without worrying about individual installations.

**Proposed Solution**
The proposed solution is to create a Docker image that encapsulates all the necessary dependencies and configurations required for Synchly. The image should allow users to interact with the tool by executing the container and utilizing the mounted volume for data syncing.

**Steps to Achieve**
Dockerfile Creation: Develop a Dockerfile that includes the Synchly tool and all required dependencies.
Volume Mounting: Enable users to mount volumes within the container to facilitate reading configuration files.

**Expected Outcome**
The expected outcome is a Docker image that simplifies the usage of Synchly, reducing the setup overhead for users and ensuring a smoother experience.

**Comment by hariprasanths** [comment]
I agree that the dockerization of synchly would be a great enhancement. But below are the few pain points that I can think of
1. Synchly would still need to communicate with the database to take the dump/backup
2. If the database is running as a separate container, synchly would need to communicate with the corresponding database containers via http, but this'll work only if the database containers have exposed their backup solutions via http

**Comment by umer2001** [comment]
@hariprasanths  have a look at #25

**Comment by umer2001** [comment]
@hariprasanths below is the guide for testing with docker:

### 1. Start the Synchly Container

Run the following command to start the Synchly container in the background:

```bash
docker run -d \
    -v /:/app/subsystem \
    --name synchly_container \
    umer2001/synchly:latest
```

This command starts the container and mounts your root directory (/) as /app/subsystem within the container.

### 2. Run Configuration Commands

Use docker exec to execute configuration commands or any other desired commands within the Synchly container. For instance:

```bash
docker exec synchly_container <command>
# example
docker exec synchly_container synchly --config=db --file /workspaces/synchly/configs/postgres.json
```

Replace `<command>` with your specific Synchly command.

### 3. Post-Configuration Commands

**IMPORTANT:** After creating or updating a configuration file, execute the following command to ensure Synchly is running:

```bash
docker exec -d synchly_container synchly --start
```

This command restarts Synchly to apply any new configurations or changes made

**Comment by umer2001** [comment]
@hariprasanths , are there bro?
